arangod/Aql/ExecutionBlock.cpp

```diff
@@ -137,7 +137,6 @@ void ExecutionBlock::traceExecuteBegin(AqlCallStack const& stack,
   // add timing for block in case profiling is turned on.
   if (_profileLevel >= ProfileLevel::Blocks) {
     // only if profiling is turned on, get current time
-    TRI_ASSERT(_startOfExecution < 0.0);
```

Comment:
Actually no. The problem is not that it can be called multiple times with different client ids, because each `traceExecuteBegin` call is matched with a `traceExecuteEnd` call _before_ the next begin call. However, `traceExecuteEnd` is _not_ called if `executeWithoutTrace` throws an exception. In that case `_startOfExecution` is not reset to `-1` and a subsequent call would than cause the assertion to fail.

Reply:
I see, thanks!